sky/execution.py

```diff
@@ -377,25 +378,44 @@ def spot_launch(
               f'{colorama.Style.RESET_ALL}')
         return
 
-    # TODO(zhwu): Refactor the Task (as Resources), so that we can enforce the
-    # following validations.
-    # Check the file mounts in the task.
-    # Disallow all local file mounts (copy mounts).
+    # Translate the workdir and local file mounts to cloud file mounts.
+    new_storage_mounts = {}
```

Comment:
One thing to think about would be how are ctrl+c/failure events handled for these newly created storage objects. For example, consider the case when a user ctrl+c `sky spot launch` after the storage has been uploaded but before the spot controller is launched. Correct me if I am wrong, but is there a chance the storage object would not get automatically deleted since teardown might not be run?

The good news is it will still be logged in `sky storage ls`, but we may have to point users to manually delete these objects. #1028 might also reduce chances of this happening.

Reply:
That is a great point! Any errors during the run could cause the storage leakage mentioned. Another problem here would be even the spot job finished and removed the storage, the `sky storage ls` on the laptop will still contain that storage. 
We may want to have a `--refresh` option for the `sky storage ls` as well.